android/src/main/kotlin/co/doneservices/callkeep/CallKeepNotificationManager.kt

```diff
@@ -0,0 +1,439 @@
+                        channel.importance > NotificationManagerCompat.IMPORTANCE_NONE) ||
+                Build.VERSION.SDK_INT < Build.VERSION_CODES.O
+    }
+
+    private fun createNotificationChanel(
+        incomingCallChannelName: String,
+        missedCallChannelName: String,
+    ) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            var channelCall = getNotificationManager().getNotificationChannel(NOTIFICATION_CHANNEL_ID_INCOMING)
+            if (channelCall != null) {
+                channelCall.setSound(null, null)
+            } else {
+                channelCall = NotificationChannel(
+                        NOTIFICATION_CHANNEL_ID_INCOMING,
+                        incomingCallChannelName,
+                        NotificationManager.IMPORTANCE_HIGH
+                ).apply {
+                    description = ""
+                    vibrationPattern =
+                            longArrayOf(0, 1000, 500, 1000, 500)
+                    lightColor = Color.RED
+                    enableLights(true)
+                    enableVibration(true)
+                    setSound(null, null)
+                }
+            }
+            channelCall.lockscreenVisibility = Notification.VISIBILITY_PUBLIC
+
+            channelCall.importance = NotificationManager.IMPORTANCE_HIGH
+
+            getNotificationManager().createNotificationChannel(channelCall)
+
+            val channelMissedCall = NotificationChannel(
+                    NOTIFICATION_CHANNEL_ID_MISSED,
+                    missedCallChannelName,
+                    NotificationManager.IMPORTANCE_DEFAULT
+            ).apply {
+                description = ""
+                vibrationPattern = longArrayOf(0, 1000)
+                lightColor = Color.RED
+                enableLights(true)
+                enableVibration(true)
+            }
+            channelMissedCall.importance = NotificationManager.IMPORTANCE_DEFAULT
+            getNotificationManager().createNotificationChannel(channelMissedCall)
+        }
+    }
+
+    private fun getAcceptPendingIntent(id: Int, data: Bundle): PendingIntent {
+        val intent = context.packageManager.getLaunchIntentForPackage(context.packageName)?.cloneFilter()
+        intent?.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+        if (intent != null) {
+            val intentTransparent = TransparentActivity.getIntentAccept(context, data)
+            return PendingIntent.getActivities(
+                    context,
+                    id,
+                    arrayOf(intent, intentTransparent),
+                    getFlagPendingIntent()
+            )
+        } else {
+            val acceptIntent = CallKeepBroadcastReceiver.getIntentAccept(context, data)
+            return PendingIntent.getBroadcast(
+                    context,
+                    id,
+                    acceptIntent,
+                    getFlagPendingIntent()
+            )
+        }
+    }
```

Comment:
I have no idea what this is o_O

Reply:
This is a mediatory activity AFAIK, but I'll dive deeper and get back to you!

Reply:
This checks if we can launch the app through `getLaunchIntentForPackage`, if the intent is not `null` then we go through with it launching our app with transparent activity, otherwise we send it to the broadcast receiver.